### PR TITLE
Closed Captions

### PR DESCRIPTION
**Pull request by msya**
Added the text tracking connection on a video and the ability to get text tracks.

### CODE REVIEW

File: vimeo-networking/src/main/java/com/vimeo/networking/model/TextTrackList.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.vimeo.networking.model;
+
+import com.vimeo.stag.UseStag;
+
+/**
+ * Created by Mohit Sarveiya on 6/29/17.
+ */
+@UseStag
+@SuppressWarnings("unused")
+public class TextTrackList extends BaseResponseList<TextTrack> {
```
Comment (anthonycr) [inline vimeo-networking/src/main/java/com/vimeo/networking/model/TextTrackList.java]: needs a unit test to verify that the stag type adapter was generated

File: vimeo-networking/src/main/java/com/vimeo/networking/model/TextTrackList.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.vimeo.networking.model;
+
+import com.vimeo.stag.UseStag;
+
+/**
+ * Created by Mohit Sarveiya on 6/29/17.
+ */
+@UseStag
```
Comment (anthonycr) [inline vimeo-networking/src/main/java/com/vimeo/networking/model/TextTrackList.java]: prefer `@UseStag(FieldOption.SERIALIZED_NAME)` to be more explicit

File: vimeo-networking/src/main/java/com/vimeo/networking/model/TextTrack.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+import org.jetbrains.annotations.Nullable;
+
+import java.io.Serializable;
+import java.util.Date;
+
+/**
+ * Created by Mohit Sarveiya on 6/29/17.
+ */
+@UseStag
```
Comment (anthonycr) [inline vimeo-networking/src/main/java/com/vimeo/networking/model/TextTrack.java]: prefer `@UseStag(FieldOption.SERIALIZED_NAME)` to be more explicit

File: vimeo-networking/src/main/java/com/vimeo/networking/model/TextTrackList.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package com.vimeo.networking.model;
+
+import com.vimeo.stag.UseStag;
+
+/**
+ * Created by Mohit Sarveiya on 6/29/17.
```
Comment (anthonycr) [inline vimeo-networking/src/main/java/com/vimeo/networking/model/TextTrackList.java]: Could use some actual documentation, even if it's minor.

File: vimeo-networking/src/main/java/com/vimeo/networking/model/TextTrack.java [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+import com.google.gson.annotations.SerializedName;
+import com.vimeo.stag.UseStag;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.io.Serializable;
+import java.util.Date;
+
+/**
+ * Created by Mohit Sarveiya on 6/29/17.
```
Comment (anthonycr) [inline vimeo-networking/src/main/java/com/vimeo/networking/model/TextTrack.java]: Could use some actual documentation, even if it's only minor.

File: vimeo-networking/src/main/java/com/vimeo/networking/model/TextTrack.java [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+ */
+@UseStag
+@SuppressWarnings({"unused", "WeakerAccess"})
+public class TextTrack implements Serializable {
```
Comment (anthonycr) [inline vimeo-networking/src/main/java/com/vimeo/networking/model/TextTrack.java]: Needs a unit test to verify that the stag type adapter was generated for this class.

File: vimeo-networking/src/main/java/com/vimeo/networking/model/Video.java [hunk 7]
```diff
@@ -520,12 +527,14 @@ public Connection getRelatedConnection() {
         return null;
     }
 
-    public int likeCount() {
-        if (getLikesConnection() != null) {
-            return getLikesConnection().mTotal;
+    @Nullable
+    public Connection getTextTracksConnection() {
+        if ((mMetadata != null) && (mMetadata.mConnections != null) && (mMetadata.mConnections.mTextTracks != null)) {
```
Comment (anthonycr) [inline vimeo-networking/src/main/java/com/vimeo/networking/model/Video.java]: Minor, you don't need all the extra parentheses

**Comment by msya** [comment]
@anthonycr Thanks for the review. I made the changes. Back to you.

**Pull request merged**